### PR TITLE
Feature/sing-in-restore-password

### PR DESCRIPTION
**Pull request by milkavladislav**

### CODE REVIEW

File: src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -82,4 +89,54 @@ export class AuthService {
...
+
+    if (user) {
+      const userAbout = await this.signUser(user.id, user.email);
+
+      const result = await this.forgotPasswordRepo.save({
+        user: user,
+        link: userAbout.token,
+      });
+
+      console.log(result);
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: delete all consols before create pr

File: src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -82,4 +89,54 @@ export class AuthService {
...
+
+      const url = `http://localhost:3000/restore-password?token=${userAbout.token}`;
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: never hardcode url in the code. move it to env or to constants

File: src/modules/auth/auth.service.ts [hunk 3]
```diff
@@ -82,4 +89,54 @@ export class AuthService {
...
+  async restorePassword({ password, token }: AuthRestorePasswordDto) {
+    const forgotPassword = await this.forgotPasswordRepo.findOne({
+      where: {
+        link: token,
+      },
+      relations: {
+        user: true,
+      },
+    });
+    const salt = genSaltSync(10);
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: 10 is magic number, never use magic number. move it to the constant and set appropriate name

File: src/modules/auth/auth.service.ts (hunk 1 above)
Comment (milkavladislav) [reply src/modules/auth/auth.service.ts]: done

File: src/modules/auth/auth.service.ts (hunk 3 above)
Comment (milkavladislav) [reply src/modules/auth/auth.service.ts]: done

File: src/modules/auth/auth.service.ts (hunk 2 above)
Comment (milkavladislav) [reply src/modules/auth/auth.service.ts]: done

File: src/modules/auth/auth.service.ts [hunk 4]
```diff
@@ -82,4 +93,53 @@ export class AuthService {
       role,
     };
   }
+
+  async forgotPassword({ email }: AuthForgotPasswordDto) {
+    const user = await this.authRepo.findOneBy({ email });
+    console.log(user);
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: delete

Comment (milkavladislav) [reply src/modules/auth/auth.service.ts]: already done, but forget to push it to origin

**Pull request merged**